lib/HLSL/HLMatrixLowerPass.cpp

```diff
@@ -476,7 +476,7 @@ void HLMatrixLowerPass::replaceAllUsesByLoweredValue(Instruction* MatInst, Value
       Instruction *PrevInst = dyn_cast<Instruction>(VecVal);
       if (PrevInst == nullptr) PrevInst = MatInst;
 
-      IRBuilder<> Builder(dxilutil::SkipAllocas(PrevInst->getNextNode()));
+      IRBuilder<> Builder(PrevInst->getNextNode());
```

Comment:
`PrevInst` is either the VecVal that the call uses or else the Matrix whose users will be replaced by it. Either way, placing the new VecToMat call just after it will ensure it is in the instruction space and before any uses of it. In fact, it's extremely unlikely that `SkipAllocas` actually advanced past anything. This likely is not changing any behavior.